packages/protoc-gen-es/src/protoc-gen-es-plugin.ts

```diff
@@ -477,9 +477,13 @@ function generateMessageJsonShape(f: GeneratedFile, message: DescMessage, target
               f.print(f.jsDoc(field, "  "));
               // eslint-disable-next-line no-case-declarations
               let jsonName: Printable = field.jsonName;
+              const startWithNumber = /^[0-9]/;
+              const containsSpecialChar = /[^a-zA-Z0-9_$]/;
+              const isAscii = /^[\x00-\x7F]*$/;
               if (jsonName === ""
-                || /^[0-9]/.test(jsonName)
-                || jsonName.indexOf("@") > -1) {
+                || startWithNumber.test(jsonName)
+                || containsSpecialChar.test(jsonName)
+                || isAscii.test(jsonName)) {
```

Comment:
Because the first regex ensures a subset of ASCII, we don't need the ASCII check. Removed it in https://github.com/bufbuild/protobuf-es/pull/1004/commits/f095a7d42140570cab7e3c749ffa3b29c3946bd1